core/src/main/java/io/temporal/samples/earlyreturn/TransactionWorkflowImpl.java

```diff
@@ -60,13 +60,14 @@ public TxResult processTransaction(TransactionRequest txRequest) {
 
   @Override
   public TxResult returnInitResult() {
```

Comment:
I think that the early return update should return a distinct result type from the workflow. In my samples I've been imagining that the early return type is a "confirmation" or "validation" of the transaction, whereas the final workflow result would be some sort of statement of the final outcome of the transaction (I'm not an expert in this domain, but perhaps the exact amount transferred is not known in advance, e.g. due to market fluctuations/exchange rates etc)

Reply:
This PR is for some logging enhancements to an already-accepted sample. It sounds like we're going back to the drawing board somewhat on what the sample should do?

For clarity, currently the outcome of the early-return is for example:
```
      {
        "transactionId": "TXID8524360451",
        "status": "Initialization successful"
      }
```
(the update mints a transaction id such as TXID8524360451)

And the workflow result is:
```
{
  "transactionId": "TXID8524360451",
  "status": "Transaction completed successfully."
}
```

@dandavison Please tell me how do better express the 'validation' vs 'final confirmation' step and I will look at making the changes

Reply:
We can leave them until a subsequent PR; we may still be making some adjustments to the API. I think one underlying issue is that the Go sample doesn't have a return type for the update. So yes, I think it would be clearest for users if the update and the workflow return distinct types, e.g. the update should return `TransactionConfirmation(transactionId, confirmationCode, ...)` and the workflow should return `TransactionStatement(transactionId, finalAmount, ...)`. But yes, don't worry, I can make the changes, and I'll talk to @drewhoskins-temporal to de-naiveify the financial concepts.

Reply:
It sounds like if I introduce a new object e.g. TxInitiationResult and make it return some stuff that feels very "this has been successfully initialized!" and keep the TxResult object for the workflow result that would satisfy.

I want to confirm this though, as I'm re-hashing a sample that was already accepted as-is and I'm trying to avoid endless churn on this. Thanks

Reply:
That sounds good. I'll get confirmation and post back here in an hour or less.

Reply:
Confirmed, @drewhoskins-temporal and I agree that distinct return types are desirable. But like I say, if you want to leave it then I can do it as I suspect there's going to be a tweak to the API coming soon.